src/nvim/tui/input.c

```diff
@@ -294,13 +294,13 @@ static void forward_simple_utf8(TermInput *input, TermKeyKey *key)
   while (*ptr) {
     if (*ptr == '<') {
       len += (size_t)snprintf(buf + len, sizeof(buf) - len, "<lt>");
+      assert(len < sizeof(buf));
     } else {
       buf[len++] = *ptr;
     }
     ptr++;
```

Comment:
I think it's better to put it somewhere reachable in both branches
```suggestion
    } else {
      buf[len++] = *ptr;
    }
    assert(len < sizeof(buf));
    ptr++;
```